controllers/v1beta1/build_qoshandler.go

```diff
@@ -52,18 +51,42 @@ func (r *LagoonBuildReconciler) whichBuildNext(ctx context.Context, opLog logr.L
 	if err := r.List(ctx, runningBuilds, listOption); err != nil {
 		return fmt.Errorf("Unable to list builds in the cluster, there may be none or something went wrong: %v", err)
 	}
+	buildsToStart := r.BuildQoS.MaxBuilds - len(runningBuilds.Items)
 	if len(runningBuilds.Items) >= r.BuildQoS.MaxBuilds {
 		// if the maximum number of builds is hit, then drop out and try again next time
 		if r.EnableDebug {
 			opLog.Info(fmt.Sprintf("Currently %v running builds, no room for new builds to be started", len(runningBuilds.Items)))
 		}
+		go r.processQueue(ctx, opLog, buildsToStart, true)
 		return nil
 	}
-	buildsToStart := r.BuildQoS.MaxBuilds - len(runningBuilds.Items)
 	if buildsToStart > 0 {
 		opLog.Info(fmt.Sprintf("Currently %v running builds, room for %v builds to be started", len(runningBuilds.Items), buildsToStart))
 		// if there are any free slots to start a build, do that here
-		listOption = (&client.ListOptions{}).ApplyOptions([]client.ListOption{
+		go r.processQueue(ctx, opLog, buildsToStart, false)
+	}
+	return nil
+}
+
+var runningProcessQueue bool
+
+// this is a processor for any builds that are currently `queued` status. all normal build activity will still be performed
+// this just allows the controller to update any builds that are in the queue periodically
+// if this ran on every single event, it would flood the queue with messages, so it is restricted using `runningProcessQueue` global
+// to only run the process at any one time til it is complete
```

Comment:
might be useful here to give a rundown of the arguments - specifically "buildsToStart" and "limitHit"